core/src/test/java/greencity/controller/HabitControllerTest.java

```diff
@@ -0,0 +1,279 @@
+
+    @InjectMocks
+    private HabitController habitController;
+
+    @Mock
+    private HabitService habitService;
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private TagsService tagsService;
+
+    @Mock
+    private ObjectMapper objectMapper;
+
+    @Mock
+    private ModelMapper modelMapper;
+
+    @Mock
+    private Principal principal;
+
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    public void setUp() {
+        this.mockMvc = MockMvcBuilders.standaloneSetup(this.habitController)
+                .setCustomArgumentResolvers(new PageableHandlerMethodArgumentResolver(),
+                        new UserArgumentResolver(this.userService, this.modelMapper))
+                .setControllerAdvice(new CustomExceptionHandler(new DefaultErrorAttributes(),
+                        this.objectMapper))
+                .build();
+    }
+
+    @Test
+    void getHabitById_WithValidId_StatusOk() throws Exception {
+        //given
+        Long habitId = 1L;
+
+        //when
+        this.mockMvc.perform(get("/habit/{id}", habitId)
+                        .locale(Locale.ENGLISH)
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andDo(print());
+
+        //then
+        verify(this.habitService).getByIdAndLanguageCode(habitId, Locale.ENGLISH.getLanguage());
+    }
+
+    @Test
+    void getAllHabits_WithValidPageAndSize_StatusOk() throws Exception {
+        //given
+        UserVO userVO = ModelUtils.getUserVO();
+        String userEmail = userVO.getEmail();
+        int pageNumber = 1;
+        int pageSize = 10;
+        Pageable pageable = PageRequest.of(pageNumber, pageSize);
+
+        //when
+        when(this.userService.findByEmail(anyString())).thenReturn(userVO);
+        this.mockMvc.perform(get("/habit")
+                .param("page", String.valueOf(pageNumber))
+                .param("size", String.valueOf(pageSize))
+                .locale(Locale.ENGLISH)
+                .principal(() -> userEmail)
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andDo(print());
+
+        //when
+        verify(this.userService).findByEmail(userEmail);
+        verify(this.habitService).getAllHabitsByLanguageCode(userVO, pageable, Locale.ENGLISH.getLanguage());
+    }
+
+    @Test
+    void getShoppingListItems_ByHabitId_StatusOk() throws Exception {
+        //given
+        Long habitId = 1L;
+
+        //when
+        this.mockMvc.perform(get("/habit/{id}/shopping-list", habitId)
+                .locale(Locale.ENGLISH)
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andExpect(result -> {
+                    String responseBody = result.getResponse().getContentAsString();
+                    String contentType = result.getResponse().getContentType();
```

Comment:
Please, remove System.out.println. They are redundant here.
And all this ".andExpected(...)" part. It does nothing useful.

Reply:
Thx done